### PR TITLE
873 extract value bar component

### LINKED ISSUE #873: Extract ValueBar component
The `ValueBar` is now created in three places ( StockDistribution, RequestStoreStats, ResponseStoreStats ).
They might not be exactly the same, but are essentially copies of the first one. It would be good to lift these out to common to enable reuse.

### PR DESCRIPTION
**Pull request by guatemartin**
Fixes #873

# 👩🏻‍💻 What does this PR do? 
 - Extracts the ValueBar component from StockDistribution, RequestStoreStats, ResponseStoreStats  in client/packages/host, and moved it to /common/src/ui/components/charts/
 - An option has been added to include or remove the starting and trailing Divider as this was also floating around in each of the components where it was before.  Defaults are leave off Divider at the start, and include Divider at the end of ValueBar.
 - Adds Storybook for Charts

# 🧪 How has/should this change been tested? 
- No functional logic was changed.
- Tests are visual. i.e navigating to Replenishment>Internal Orders, selecting an order with "Draft" status.  Find an order with detail lines where AMC > 0.

## 💌 Any notes for the reviewer?
- Not sure my naming convention of "startDivider" and "endDivider" are very descriptive in the ValueBar component.

## 📃 Documentation
- The strength of this component is when 2 or more ValueBar components are put together in a Flexbox.  I wasn't sure of the best way to convey that in StoryBook, or anywhere else for that matter.

### CODE REVIEW

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:58]: should have the `endDivider` too - it was tucked away in the component 🙄 
which means that the other two instances below also need the end divider.

Comment (mark-prins) [reply client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:58]: actually - the behaviour differs: if the first ValueBar has no value, it isn't rendered, which means that the first divider isn't shown. In the original version, the divider is shown regardless. I suggest updating the ValueBar, so that the start divider is shown regardless of whether the rest of the component is shown or not.

File: client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx
Comment (mark-prins) [inline client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx:30]: The previous behaviour rendered the start divider regardless - something like this would do it:
```suggestion
  if (value === 0) return startDivider ? <Divider /> : null;
```

**Review by mark-prins — CHANGES_REQUESTED** [review]
🎉  Looking great! and the bundle size is reduced which is fantastic!

I have a small suggestion - because the divider isn't shown in some cases. 
And the prop names of `endDivider` and `startDivider` are fine by me - thanks

File: client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx
Comment (guatemartin) [reply client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx]: That makes sense.  Commit submitted.  Cheers.

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx
Comment (guatemartin) [reply client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:58]: My apologies, I missed this one.  Will rectify.

File: client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx
Comment (guatemartin) [reply client/packages/common/src/ui/components/charts/ValueBar/ValueBar.tsx]: I've just noticed on one of the "ResponseStoreStats" example, the Dividers don't show at all (and labels are ugly).
They are there, they just have a width of 0.  It has to do with the way 'requestedPercent' is calculated for the width of the div.  Not sure if that should be tackled here in this PR or not.

Screenshot 2023-03-30 122659

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx
Comment (mark-prins) [reply client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:58]: oops 🙄  just realised that `endDivider` defaults to true! Ignore previous comment..

**Review by mark-prins — APPROVED** [review]
👍 looks great thanks!
we usually leave it to the author of the pr to merge once approved.. just in case there's a reason to hold off. Since this is going to the develop branch, you can merge whenever.

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx
Comment (guatemartin) [reply client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:58]: No worries.  It did make me think about it a bit more though.

**Pull request merged**
